playwright/e2e/spaces/threads-activity-centre/index.ts

```diff
@@ -0,0 +1,355 @@
+/*
+Copyright 2024 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { JSHandle, Locator, Page } from "@playwright/test";
+
+import type { MatrixEvent, IContent, Room } from "matrix-js-sdk/src/matrix";
+import { test as base, expect } from "../../../element-web-test";
+import { Bot } from "../../../pages/bot";
+import { Client } from "../../../pages/client";
+import { ElementAppPage } from "../../../pages/ElementAppPage";
+
+/**
+ * Set up for a read receipt test:
+ * - Create a user with the supplied name
+ * - As that user, create two rooms with the supplied names
+ * - Create a bot with the supplied name
+ * - Invite the bot to both rooms and ensure that it has joined
+ */
+export const test = base.extend<{
+    roomAlphaName?: string;
+    roomAlpha: { name: string; roomId: string };
+    roomBetaName?: string;
+    roomBeta: { name: string; roomId: string };
+    msg: MessageBuilder;
+    util: Helpers;
+}>({
+    displayName: "Mae",
+    botCreateOpts: { displayName: "Other User" },
+
+    roomAlphaName: "Room Alpha",
+    roomAlpha: async ({ roomAlphaName: name, app, user, bot }, use) => {
+        const roomId = await app.client.createRoom({ name, invite: [bot.credentials.userId] });
+        await use({ name, roomId });
+    },
+    roomBetaName: "Room Beta",
+    roomBeta: async ({ roomBetaName: name, app, user, bot }, use) => {
+        const roomId = await app.client.createRoom({ name, invite: [bot.credentials.userId] });
+        await use({ name, roomId });
+    },
+    msg: async ({ page, app, util }, use) => {
+        await use(new MessageBuilder(page, app, util));
+    },
+    util: async ({ roomAlpha, roomBeta, page, app, bot }, use) => {
+        await use(new Helpers(page, app, bot));
+    },
+});
+
+/**
+ * A utility that is able to find messages based on their content, by looking
+ * inside the `timeline` objects in the object model.
+ *
+ * Crucially, we hold on to references to events that have been edited or
+ * redacted, so we can still look them up by their old content.
+ *
+ * Provides utilities that build on the ability to find messages, e.g. replyTo,
+ * which finds a message and then constructs a reply to it.
+ */
+export class MessageBuilder {
```

Comment:
This is basically a lighter version of the read-receipt MessageBuilder.
Copy paste, added mention in thread and removed all the unused parts.